### PR TITLE
PrecompileTools

### PR DESCRIPTION
**Pull request by LuEdRaMo**
In order to solve https://github.com/PerezHz/PlanetaryEphemeris.jl/issues/13, this PR uses `PrecompilationTools.jl` to precompile the `propagate` function. With this PR, the package precomilation takes ∼44 minutes, but TTFX is significantly reduced.

### CODE REVIEW

**Comment by PerezHz** [comment]
Hey, any performance gains that we can get are very well received, many thanks! Is this ready for merging?

File: src/precompile.jl [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
     jd0 = datetime2julian(DateTime(2000,1,1,12)) 
     # Number of years 
     nyears = 2031.0 - year(julian2datetime(jd0))
-
-    @precompile_all_calls begin
-        propagate(1, jd0, nyears, Val(false); dynamics = DE430!, order = 25, abstol = 1.0E-20)
+    @compile_workload begin
+        # All calls in this block will be precompiled, regardless of whether
+        # they belong to your package or not (on Julia 1.8 and higher)
         propagate(1, jd0, nyears, Val(true); dynamics = DE430!, order = 25, abstol = 1.0E-20)
```
Comment (lbenet) [inline src/precompile.jl:11]: Does it change anything to use `propagate` for a two-step integration?

**Review by lbenet — APPROVED** [review]
Thanks a lot @LuEdRaMo. This PR LGTM, so I am in favor of merging it, and tagging a new (minor) version. The comment I left is optional, and perhaps useless.

File: src/precompile.jl (hunk 1 above)
Comment (LuEdRaMo) [reply src/precompile.jl:11]: Integrating two steps changes precompilation time by a few minutes. I've noted that the actual integration represents only a small percentage of precompilation time. This PR is a solution to https://github.com/PerezHz/PlanetaryEphemeris.jl/issues/13, but we should investigate further why the package takes so long precompilating. If you agree I will bump the patch version and merge...

File: src/constants.jl [hunk 2]
```diff
@@ -5,7 +5,7 @@ const PE = PlanetaryEphemeris
 const src_path = dirname(pathof(PlanetaryEphemeris))
 
 # Integration parameters
-const order = 30
+const order = 25
```
Comment (lbenet) [inline src/constants.jl:8]: Is this change needed?

File: src/precompile.jl [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+    # Maximum number of steps
+    maxsteps = 2
     # Starting time of integration
     jd0 = datetime2julian(DateTime(2000,1,1,12)) 
     # Number of years 
     nyears = 2031.0 - year(julian2datetime(jd0))
+    # Dense output
+    dense = Val(true)
+    # Dynamical function 
+    dynamics = DE430!
```
Comment (lbenet) [inline src/precompile.jl]: Should these constants be `local`s?

File: src/constants.jl (hunk 2 above)
Comment (LuEdRaMo) [reply src/constants.jl:8]: All integrations that we have done with PE are with 25 order polynomias, so I think this is more consistent.

File: src/precompile.jl (hunk 3 above)
Comment (LuEdRaMo) [reply src/precompile.jl]: Actually, they are treated as globals, so I think I should add `const`.

**Comment by lbenet** [comment]
There are issues to bump the new version of NEOs; see [this comment](https://github.com/JuliaRegistries/General/pull/84182#issuecomment-1562076302). So I suggest to check this PR, merge it and bump the new version as soon as possible....

**Comment by PerezHz** [comment]
> There are issues to bump the new version of NEOs; see https://github.com/JuliaRegistries/General/pull/84182#issuecomment-1562076302. So I suggest to check this PR, merge it and bump the new version as soon as possible....

I went ahead and updated the Project.toml of NEOs.jl, this should help solve the automerge issues


[see here](https://github.com/PerezHz/NEOs.jl/commit/d2d75e3378e2b5fc63621eedb1236865e4f951c2#commitcomment-115023387)

**Comment by lbenet** [comment]
Thanks! I think we will still need some further packages upgrades, e.g., IntervalArithmetic, which are somehow blocked by PlanetaryEphemeris. Once this PR is ready, we can go further there...

**Comment by PerezHz** [comment]
> Thanks! I think we will still need some further packages upgrades, e.g., IntervalArithmetic, which are somehow blocked by PlanetaryEphemeris. Once this PR is ready, we can go further there...

I added the `const` where @LuEdRaMo suggested to add them; that was the last suggestion standing, so I'll go ahead and merge this PR; again many thanks @LuEdRaMo!

Also, just for completeness I'll mention that NEOs.jl v0.3.0 was just released, I think we can address the packages upgrades in later PRs...
